### DISCUSSION TITLE
Connection closing with Windows 10 VPN client

### DISCUSSION THREAD

**Discussion by DavorJ** [strongswan/strongswan#1029] [Q&A] [answered]
I am running a web-video steaming host behind the VPN gateway. The road-warrior connects to the VPN, and is able to access the host and its web stream through the gateway.

This works very well from Android clients, but from Windows 10 clients the stream will stop after about 5 minutes, as if the connection times out. Refreshing the webpage will allow the stream to continue (i.e. no need to reconnect the VPN).

I am puzzled as to what could be causing the disconnects? Below part of my `swanctl.conf`. Any ideas?

```
connections {
   roadwarrior {
      version = 2
      local_addrs = %any
      proposals = aes256-sha1-prfsha1-modp4096-modp2048
      dpd_delay = 300s
      # Time to schedule IKE rekeying. 0 = disabled. 4h = default.
      # Win 7 rekeys after 3h, Win 10 after 7.6h.
      # Win 7 does not like a VPN gateway to take the initiative.
      rekey_time = 12h
      # Enforce UDP encapsulation by faking NAT-D payloads.
      encap = yes
      pools = dhcp
      local {
         auth = pubkey
         ...
      }
      remote {
         auth = eap-mschapv2
         ...
      }
      children {
         net {
            esp_proposals = aes256-sha1
            local_ts = xxx.xxx.xxx.xxx/xx
         }
      }
   }
}
```

**Comment by fractalb** [comment]
`dpd_delay = 300s`
Is this the cause of 5min tunnel timeout? What if you disable  dpd on the Strongswan side?

**Reply by tobiasbrunner** [reply]
It seems unlikely that DPD causes this. But try reading the log at the time of the disconnect.

**Reply by DavorJ** [reply]
I checked with `journalctl -u strongswan-swanctl.service -b -f`.

After the VPN connection is established and stream is started, absolutely nothing happens until the stream stops. I played with disabling `dpd_delay` and `encap`, but it had no effect.

I keep thinking it is a Windows firewall thing: after buffering there may be no communication for a time. But I wonder, shouldn't charon keep-alives kick in in that situation?

**Reply by tobiasbrunner** [reply]
> But I wonder, shouldn't charon keep-alives kick in in that situation?

If you are referring to DPDs, then these are sent if no IKE or ESP messages are received within `dpd_delay`. The IKE messages you should see in the log, for ESP traffic check the traffic counters.

**Comment by DavorJ** [comment, marked as answer]
Ok, this is mine stupidity. I had in fact `esp_proposals = aes256-sha256-sha1-modp2048`. Win10 client always succeded in making the connection and seemed to work fine for a while, but stopped working after some time. I noticed during debugging when I set the `inactivity = 120s` that the ESP proposals didn't match in the logs. Otherwise, logs showed nothing.

Moral of the story: make sure to follow [Windows Clients](https://docs.strongswan.org/docs/5.9/interop/windowsClients.html) to the letter in setting IKE / ESP proposals.
